File: src/main/java/client/renderables/FPSViewer.java
package client.renderables;

import client.Game;
import client.renderables.Drawable;
import client.renderables.Entity;

import java.awt.*;

public class FPSViewer extends Entity implements Drawable {
    private int ticks;
    private int frames;

    private int averageTicks;
    private int averageFrames;

    public void draw (Graphics g) {
        frames++;
        if (Game.f3) {
            g.setColor(Color.GREEN);
            g.setFont(new Font(null, 0, 24));
            g.drawString("FPS: " + averageFrames + " Ticks: " + averageTicks, 1280 - (14 * (5 + 2 + 6 + 2)), 24);
        }
    }

    public void tick () {
        ticks ++;
    }

    public void second () {
        averageFrames = frames;
        averageTicks = ticks;
        frames = 0;
        ticks = 0;
    }
}

File: src/main/java/client/renderables/FPSViewer.kt
package client.renderables

import java.awt.Color
import java.awt.Font
import java.awt.Graphics

//TODO: This shouldn't be an entity.
class FPSViewer : RenderedEntity(), Drawable {
    private var ticks: Int = 0
    private var frames: Int = 0

    private var averageTicks: Int = 0
    private var averageFrames: Int = 0

    override fun draw(g: Graphics) {
        frames++
        if (true) {
            g.color = Color.GREEN
            g.font = Font(null, 0, 24)
            g.drawString("FPS: $averageFrames Ticks: $averageTicks", 1280 - 14 * (5 + 2 + 6 + 2), 24)
        }
    }

    override fun tick() {
        ticks++
    }

    override fun second() {
        averageFrames = frames
        averageTicks = ticks
        frames = 0
        ticks = 0
    }
}
